src/test/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNodeTest.java

```diff
@@ -0,0 +1,155 @@
+                .isEqualTo(11);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(-11);
+    }
+
+    @Test
+    public void testCompareToSameLinesAndColumns() {
+        final MockNode node1 = new MockNode(10, 12, "abc");
+        final MockNode node2 = new MockNode(10, 12, "def");
+
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(-3);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(3);
+    }
+
+    @Test
+    public void testCompareToIdentical() {
+        final MockNode node1 = new MockNode(10, 12, "abc");
+        final MockNode node2 = new MockNode(10, 12, "abc");
+
+        assertWithMessage("Invalid comparison result")
+                .that(node1.compareOrder(node2))
+                .isEqualTo(0);
+        assertWithMessage("Invalid comparison result")
+                .that(node2.compareOrder(node1))
+                .isEqualTo(0);
+    }
+
+    /* default */ private static class MockNode extends AbstractNode {
+
+        private final int lineNumber;
+        private final int columnNumber;
+        private final String localPart;
+
+        /* default */ MockNode(int lineNumber, int columnNumber) {
+            this(lineNumber, columnNumber, "");
+        }
+
+        /* default */ MockNode(int lineNumber, int columnNumber, String localPart) {
+            super(new GenericTreeInfo(Configuration.newConfiguration()));
+            this.lineNumber = lineNumber;
+            this.columnNumber = columnNumber;
+            this.localPart = localPart;
+        }
+
+        @Override
+        public int getLineNumber() {
+            return lineNumber;
+        }
+
+        @Override
+        public int getColumnNumber() {
+            return columnNumber;
+        }
+
+        @Override
+        public int getTokenType() {
+            return 0;
+        }
+
+        @Override
+        public DetailAST getUnderlyingNode() {
+            return null;
+        }
+
+        @Override
+        public int getNodeKind() {
+            return 0;
+        }
+
+        @Override
+        public String getStringValue() {
+            throw throwUnsupportedOperationException();
+        }
+
+        @Override
+        public String getLocalPart() {
+            return localPart;
+        }
+
+        @Override
+        public NodeInfo getParent() {
+            return null;
+        }
+
+        @Override
+        public String getAttributeValue(String s, String s1) {
+            return null;
+        }
+
+        @Override
+        public NodeInfo getRoot() {
+            return null;
+        }
+
+        private static UnsupportedOperationException throwUnsupportedOperationException() {
+            return new UnsupportedOperationException("Operation is not supported");
```

Comment:
method does not throw ... it creates object.  Please rename to `createU........`

Reply:
done